File: prolanlab-java/src/main/java/com/prolanlab/tutorials/TutorialModule.java
package com.prolanlab.tutorials;

import com.prolanlab.tutorials.java.JavaTutorialController;
import com.prolanlab.tutorials.python.PythonTutorialController;
import com.prolanlab.Main;

import java.util.Scanner;

public class TutorialModule {
    private final Scanner scanner = Main.getScanner();
    
    public void start() {
        boolean inTutorialModule = true;
        
        while (inTutorialModule) {
            displayTutorialMenu();
            int choice = getUserChoice();
            
            switch (choice) {
                case 1:
                    new JavaTutorialController().start();
                    break;
                case 2:
                    new PythonTutorialController().start();
                    break;
                case 3:
                    System.out.println("Returning to main menu...");
                    inTutorialModule = false;
                    break;
                default:
                    System.out.println("Invalid option. Please try again.");
            }
            
            if (inTutorialModule && choice != 3) {
                System.out.println("\n" + "=".repeat(75));
                System.out.println("Press Enter to return to tutorial menu...");
                scanner.nextLine();
            }
        }
    }
    
    private void displayTutorialMenu() {
        System.out.println("\n" + "=".repeat(75));
        System.out.println("                    [T] TUTORIALS");
        System.out.println("=".repeat(75));
        System.out.println();
        System.out.println(" 1. Java Tutorial");
        System.out.println(" 2. Python Tutorial");  
        System.out.println();
        System.out.println(" 3. << Back to Main Menu");
        System.out.println();
        System.out.println("=".repeat(75));
        System.out.print("Enter your choice (1-3): ");
    }
    
    private int getUserChoice() {
        try {
            int choice = scanner.nextInt();
            scanner.nextLine(); // Consume the newline
            return choice;
        } catch (Exception e) {
            scanner.nextLine(); // Clear invalid input
            return -1;
        }
    }
}

File: prolanlab-python/src/prolanlab/tutorials/tutorial_module.py
"""
Tutorial Module - Main controller for all tutorials
"""

from prolanlab.tutorials.java.java_tutorial_controller import JavaTutorialController
from prolanlab.tutorials.python.python_tutorial_controller import PythonTutorialController


class TutorialModule:
    def __init__(self):
        pass

    def start(self):
        in_tutorial_module = True
        
        while in_tutorial_module:
            self._display_tutorial_menu()
            choice = self._get_user_choice()
            
            if choice == 1:
                JavaTutorialController().start()
            elif choice == 2:
                PythonTutorialController().start()
            elif choice == 3:
                print("Returning to main menu...")
                in_tutorial_module = False
            else:
                print("Invalid option. Please try again.")
            
            if in_tutorial_module and choice != 3:
                print("\n" + "=" * 75)
                print("Press Enter to return to tutorial menu...")
                input()

    def _display_tutorial_menu(self):
        print("\n" + "=" * 75)
        print("                    [T] TUTORIALS")
        print("=" * 75)
        print()
        print(" 1. Java Tutorial")
        print(" 2. Python Tutorial")
        print()
        print(" 3. << Back to Main Menu")
        print()
        print("=" * 75)
        print("Enter your choice (1-3): ", end="")

    def _get_user_choice(self):
        try:
            choice = int(input())
            return choice
        except ValueError:
            return -1